src/common.rs

```diff
@@ -70,6 +70,23 @@ cfg_if::cfg_if! {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct JsonWrapper<A: Serialize> {
```

Comment:
I'm actually not sure if `A` needs `Serialize` added explicitly or if the deriving automatically adds that. This has to be checked with someone with deeper Rust knowledge. @ueno any thoughts on this?

Reply:
Weak opinion but I think it's a little clearer to explicitly state that `A` is bound to `Serialize` at struct definition; without it one could create a `JsonWrapper` with a non-serializable `results` but this error is caught only when it's actually being serialized.

Reply:
Do we then need to add this for `Deserialize` and `Debug`?

Reply:
I agree. All the other changes LGTM

Reply:
Searching the documentation, I found that derived `Serialize` and `Deserialize` add the trait bound for generic structure fields, which is probably sufficient in our case:
https://serde.rs/attr-bound.html

Moreover, trying to apply the bounds manually in the structure proved to be complicated due to the required lifetime binding for `Deserialize`. It is probably possible, but I failed to understand how to write the code (I need to study more rust).

Searching in the internet how people solved this, I found that it is somewhat unusual to restrict the traits in the structure for generic fields. Normally such restrictions are applied in the `impl` which defines the behavior.

For all above, I think it is sufficient to leave the derived `Serialize` and `Deserialize` to take care of the traits binding and add the trait restriction to the `impl`. 

@ueno @THS-on What do you think? If you know how to enforce the trait binding in the structure, feel free to suggest.

ps.: The result seems to be that `JsonWrapper::new()` will not accept types without the `Serialize` and `Deserialize` traits, but it would still be possible to manually create a `JsonWrapper` structure containing a structure in `results` without the traits, which is basically the issue @ueno pointed out.